lib/client/presence/presence.js

```diff
@@ -56,13 +61,14 @@ Presence.prototype.destroy = function(callback) {
           }, next);
         },
         function(next) {
+          if (!presence._wantsDestroy) return next();
           async.each(remoteIds, function(presenceId, next) {
             presence._remotePresenceInstances[presenceId].destroy(next);
           }, next);
         }
       ],
       function(error) {
-        delete presence.connection._presences[presence.channel];
+        if (presence._wantsDestroy) delete presence.connection._presences[presence.channel];
         presence._callbackOrEmit(error, callback);
```

Comment:
Note that this will still invoke the `destroy()` callback, even if the destroy is cancelled. It does this because I feel like all callbacks should eventually be invoked (eg if you tried to promisify this method, it would never resolve or reject).

However, if clients care about this, we could either:
- expose `wantsDestroy` publically so they can check this in their callback; or
- pass `wantsDestroy` into the callback as an argument